honeybee_schema/radiance/lightsource.py

```diff
@@ -0,0 +1,176 @@
+"""SensorGrid and Sensor Schema"""
+from pydantic import Field, constr
+from typing import Optional
+
+from .._base import NoExtraBaseModel
+from .wea import Wea
+
+
+class _GlowHemisphere(NoExtraBaseModel):
+    """Hidden base class for Ground and SkyHemisphere."""
+
+    r_emittance: float = Field(
+        default=1.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the red channel emittance.'
+    )
+
+    g_emittance: float = Field(
+        default=1.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel emittance.'
+    )
+
+    b_emittance: float = Field(
+        default=1.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel of the emittance.'
+    )
+
+
+class Ground(_GlowHemisphere):
+    """Ground component of the sky sphere."""
+
+    type: constr(regex='^Ground$') = 'Ground'
+
+
+class SkyHemisphere(_GlowHemisphere):
+    """SkyHemisphere component of the sky sphere."""
+
+    type: constr(regex='^SkyHemisphere$') = 'SkyHemisphere'
+
+
+class SkyDome(NoExtraBaseModel):
+    """Hidden base class for all sky domes."""
+
+    type: constr(regex='^SkyDome$') = 'SkyDome'
+
+    ground_hemisphere: Optional[Ground] = Field(
+        default=Ground(),
+        description='Optional ground glow source.'
+        )
+
+    sky_hemisphere: Optional[SkyHemisphere] = Field(
+        default=SkyHemisphere(),
+        description='Optional sky hemisphere glow source.'
+        )
+
+
+class _PointInTime(SkyDome):
+    """Hidden base class for all point-in-time sky domes."""
+
+    ground_reflectance: float = Field(
+        default=0.2,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the ground reflectance.'
+    )
+
+
+class _SkyWithSun(_PointInTime):
+    """Hidden base class for all altitude/azimuth sky domes."""
+
+    altitude: float = Field(
+        ...,
+        ge=-90,
+        le=90,
+        description='The solar altitude measured in degrees above the horizon.'
```

Comment:
Is it still correct to say `above the horizon` now that we allow negative values?

Reply:
I think it's fine to say this. I'll just add a note that negative values are below the horizon.